### PR TITLE
config: add templating for enabling wc to sc probes

### PR DESCRIPTION
**Pull request by anders-elastisys**
> [!warning]
> **This is public repository, ensure not to disclose:**
> - [ ] personal data beyond what is necessary for interacting with this pull request
> - [ ] business confidential information, such as customer names

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [x] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change  
- [ ] kind/dev-change    
- [ ] kind/security      
- [ ] [kind/adr]()       

### What does this PR do / why do we need this PR?

The wc to sc blackbox probes could not be disabled and are always configured with `baseDomain`, which if wc and sc clusters have different `baseDomain` causes alerts as the probes will fail. This PR makes it possible to disable all wc to sc probes as well as setting `scDomain` and `scOpsDomain` that can be used to override `baseDomain` and `opsDomain` in cases where wc and sc domain is not the same.

- Fixes #

#### Checklist

- [x] Proper commit message prefix on all commits
  
- Change checks:
  - [x] The change is transparent
  - [ ] The change is disruptive
  - [ ] The change requires no migration steps
  - [ ] The change requires migration steps
- Metrics checks:
  - [ ] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packages in the `NetworkPolicy Dashboard`
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy

### CODE REVIEW

File: helmfile/values/prometheus-blackbox-exporter-wc.yaml.gotmpl [hunk 1]
```diff
@@ -67,32 +67,36 @@ serviceMonitor:
       module: http_2xx
     {{- end }}
 
+    {{- if .Values.prometheusBlackboxExporter.targets.sc}}
+    {{- $scDomain := default .Values.global.baseDomain .Values.global.scDomain }}
+    {{- $scOpsDomain := default .Values.global.opsDomain .Values.global.scOpsDomain }}
```
Comment (Zash) [inline helmfile/values/prometheus-blackbox-exporter-wc.yaml.gotmpl:72]: Is it me looking in the wrong place or is `default` missing from the [helmfile docs](https://helmfile.readthedocs.io/en/latest/templating_funcs/)?

Comment (anders-elastisys) [reply helmfile/values/prometheus-blackbox-exporter-wc.yaml.gotmpl:72]: I looked at how it was done [here](https://github.com/elastisys/compliantkubernetes-apps/blob/de2875d3467ea5008eca9e43d605d57fb06219c2/helmfile/values/thanos/objectstorage-secret.yaml.gotmpl#L8) and tested it out and it worked. Maybe it needs to be documented, and I am open for suggestions if there is a better or more clear way to do this.

Comment (Zash) [reply helmfile/values/prometheus-blackbox-exporter-wc.yaml.gotmpl:72]: Best case, someone forgot to document it. Worst case, it's some unofficial feature that might change behavior in a future version without notice.

Actually it seems to be an omission on that template functions page, [this page does mention it](https://helmfile.readthedocs.io/en/latest/writing-helmfile/#missing-keys-and-default-values).

Comment (anders-elastisys) [reply helmfile/values/prometheus-blackbox-exporter-wc.yaml.gotmpl:72]: It should just be go templating (and sprig templates) under the hood, and there is docs for the default function [here](http://masterminds.github.io/sprig/defaults.html)

**Review by Elias-elastisys — APPROVED** [review]
lgtm

**Pull request merged**